src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -187,25 +189,39 @@ applyCoinbase
     -> MinerInfo
     -> ParsedDecimal
     -> PublicData
+    -> BlockHash
     -> IO (CommandResult [TxLog Value])
-applyCoinbase logger dbEnv minerInfo reward pd = do
-
+applyCoinbase logger dbEnv mi@(MinerInfo mid mks) reward pd ph = do
     -- cmd env with permissive gas model
-    let cmdEnv = CommandEnv Nothing Transactional dbEnv logger freeGasEnv pd noSPVSupport
-        coinbaseReq = RequestKey $ H.toUntypedHash (H.hash "COINBASE" :: H.PactHash)
+    let cenv = CommandEnv Nothing Transactional dbEnv logger freeGasEnv pd noSPVSupport
+        initState = initCapabilities [magic_COINBASE]
 
-    resultE <- catchesPactError $ coinbase cmdEnv minerInfo reward coinbaseReq
+    -- build the pure Exec so we can build the cmd
+    let cexec = Exec $! mkCoinbaseExec mid mks reward
 
-    case resultE of
+    ccmd <- mkCommand [] (_pdPublicMeta pd) (sshow ph) cexec
 
-      Left e ->
-        jsonErrorResult cmdEnv coinbaseReq e [] (Gas 0)
-          "genesis tx failure for request key while running genesis"
+    cexec' <- case verifyCommand @PublicMeta ccmd of
```

Comment:
You should be using `applyExec'` like on #L375 instead of reproducing the `ProcSucc` stuff

Reply:
Amended